source/common/upstream/maglev_lb.cc

```diff
@@ -26,19 +31,28 @@ MaglevTable::MaglevTable(const HostVector& hosts) {
       table_[c] = hosts[i];
       table_build_entries[i].next_++;
       table_index++;
-      if (table_index == TableSize) {
+      if (table_index == table_size_) {
```

Comment:
this is partially pre-existing, but wouldn't it be more idiomatic to have a doubly nested if? Then you can tell at a glance how many iterations there are rather than having to read the loop body (modulo the permutations loop).
```
for (uint64_t table_index = 0; table_index < table_size_; ++table_index) {
  for (uint64_t i = 0; i < hosts.size(); ++i) {
    uint64_t c = permutation(table_build_entries[i]);
    while (table_[c] != nullptr) {
      table_build_entries[i].next_++;
      c = permutation(table_build_entries[i]);
    }
    table_[c] = hosts[i];
    table_build_entries[i].next_++;
  }
#ifndef NVLOG
  ENVOY_LOG(trace, "maglev: i={} host={}", i, table_[table_index]->address()->asString());
#endif
}
```

Reply:
The code you wrote above I don't think works. It needs to loop through all hosts each iteration, and handle when table size is not divisible by hosts. In general I would prefer to keep the code as is which follows the paper. IMO for this kind of thing it's easier to compare.

Reply:
I see, I got my loops garbled. Can you add a comment such as:
  // Derived from [maglev paper link] page 6, "Pesudocode 1".

Reply:
See comment at line 12.